Lib/tarfile.py

```diff
@@ -1468,6 +1468,8 @@ def __init__(self, name=None, mode="r", fileobj=None, format=None,
             if hasattr(fileobj, "mode"):
                 self._mode = fileobj.mode
             self._extfileobj = True
+
+        name = os.fspath(name) if name else None
```

Comment:
Move this line before `if not fileobj`: the code calls `os.path.exists()` and `bltn_open()` with name.

Both functions accepts`pathlib.Path` object but let's be consistent in `Path -> str` conversion, please

Reply:
@asvetlov 
Oh, finally I got It. Updated.